### ISSUE TITLE
Bold ZeroTopPaddingTextView displays cut off on 4.4

### ISSUE THREAD

**Issue by achuinard** [code-troopers/android-betterpickers#46]
Using the new 4.4, the bolded numbers don't look right.  I assume something's going on the with ZeroTopPaddingTextView.  I attached a SS that I launched from the demo app.
![screenshot_2013-11-22-01-21-28](https://f.cloud.github.com/assets/469647/1598979/f6c8cc4a-5346-11e3-9afd-e282f51c91c0.png)

**Comment by JordanMarshall** [comment]
I can confirm this (Android 4.4, Nexus 5). I am currently working around the issue by calling updatePaddingForBoldDate() on the offending ZeroTopPaddingTextView.

**Comment by niko001** [comment]
Thanks, @theshadowbelow, this fixed it for me, as well!

**Comment by nealsanche** [comment]
I also found this to be the case in HmsView.java and had to make the following modifications:

```
    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        mHoursOnes = (ZeroTopPaddingTextView) findViewById(R.id.hours_ones);
        mMinutesTens = (ZeroTopPaddingTextView) findViewById(R.id.minutes_tens);
        mMinutesOnes = (ZeroTopPaddingTextView) findViewById(R.id.minutes_ones);
        mSecondsTens = (ZeroTopPaddingTextView) findViewById(R.id.seconds_tens);
        mSecondsOnes = (ZeroTopPaddingTextView) findViewById(R.id.seconds_ones);
        if (mHoursOnes != null) {
            mOriginalHoursTypeface = mHoursOnes.getTypeface();
            mHoursOnes.updatePaddingForBoldDate();
        }
        if (mMinutesTens != null) {
            mMinutesTens.updatePaddingForBoldDate();
        }
        if (mMinutesOnes != null) {
            mMinutesOnes.updatePaddingForBoldDate();
        }
        // Set the lowest time unit with thin font (excluding hundredths)
        if (mSecondsTens != null) {
            mSecondsTens.setTypeface(mAndroidClockMonoThin);
            mSecondsTens.updatePadding();
        }
        if (mSecondsOnes != null) {
            mSecondsOnes.setTypeface(mAndroidClockMonoThin);
            mSecondsOnes.updatePadding();
        }
    }
```

**Comment by inteist** [comment]
That seemed to resolve the issue for me (had to add am/pm as well to be able to call the updapaddingforbold)

```
public void setTime(int hoursTensDigit, int hoursOnesDigit, int minutesTensDigit,
        int minutesOnesDigit) {
    if (mHoursTens != null) {
        // Hide digit
        if (hoursTensDigit == -2) {
            mHoursTens.setVisibility(View.INVISIBLE);
        } else if (hoursTensDigit == -1) {
            mHoursTens.setText("-");
            mHoursTens.setTypeface(mAndroidClockMonoThin);
            mHoursTens.setEnabled(false);
            mHoursTens.updatePadding();
            mHoursTens.setVisibility(View.VISIBLE);
        } else {
            mHoursTens.setText(String.format("%d", hoursTensDigit));
            mHoursTens.setTypeface(mOriginalHoursTypeface);
            mHoursTens.setEnabled(true);
            mHoursTens.updatePaddingForBoldDate();
            mHoursTens.setVisibility(View.VISIBLE);
        }
    }
    if (mHoursOnes != null) {
        if (hoursOnesDigit == -1) {
            mHoursOnes.setText("-");
            mHoursOnes.setTypeface(mAndroidClockMonoThin);
            mHoursOnes.setEnabled(false);
            mHoursOnes.updatePadding();
        } else {
            mHoursOnes.setText(String.format("%d", hoursOnesDigit));
            mHoursOnes.setTypeface(mOriginalHoursTypeface);
            mHoursOnes.setEnabled(true);
            mHoursOnes.updatePaddingForBoldDate();
        }
    }

    if (mAMPMLabel != null) {
        mAMPMLabel.updatePaddingForBoldDate();
    }

    if (mMinutesTens != null) {
        if (minutesTensDigit == -1) {
            mMinutesTens.setText("-");
            mMinutesTens.setEnabled(false);
        } else {
            mMinutesTens.setEnabled(true);
            mMinutesTens.setText(String.format("%d", minutesTensDigit));
        }
    }
    if (mMinutesOnes != null) {
        if (minutesOnesDigit == -1) {
            mMinutesOnes.setText("-");
            mMinutesOnes.setEnabled(false);
        } else {
            mMinutesOnes.setText(String.format("%d", minutesOnesDigit));
            mMinutesOnes.setEnabled(true);
        }
    }
}
```

**Comment by derekbrameyer** [comment]
Hi all, sorry for the late response but I have fixed this in the latest 1.5.2 release!
